src/commands/local.ts

```diff
@@ -913,9 +913,29 @@ async function handleSdk(
             throw new UserError("Could not find the SDK for the frontend.");
         }
 
+        let workspaceUrl: string | undefined;
+        if (process.env?.["GITPOD_WORKSPACE_URL"]) {
+            const gitPodWorkspaceUrl = process.env["GITPOD_WORKSPACE_URL"];
+            const insertPortIndex = gitPodWorkspaceUrl.indexOf("https://") + "https://".length;
+            workspaceUrl =
+                gitPodWorkspaceUrl.slice(0, insertPortIndex) +
+                `${options.port}-` +
+                gitPodWorkspaceUrl.slice(insertPortIndex);
```

Comment:
IMO, here is much simpler to replace `https://` with `https://${options.port}-`

The code will be shorter:
```
const url = process.env["GITPOD_WORKSPACE_URL"].
const gitPodWorkspaceUrl = url.replace(`https://`, `https://${options.port}-`);
```